src/core/mac/mac_frame.hpp

```diff
@@ -757,16 +788,25 @@ class Frame: public otRadioFrame
      * @returns A pointer to the char string buffer.
      *
      */
-    const char *ToInfoString(char *aBuf, uint16_t aSize);
+    const char *ToInfoString(char *aBuf, uint16_t aSize) const;
 
 private:
-    uint8_t *FindSequence(void);
-    uint8_t *FindDstPanId(void);
-    uint8_t *FindDstAddr(void);
-    uint8_t *FindSrcPanId(void);
-    uint8_t *FindSrcAddr(void);
-    uint8_t *FindSecurityHeader(void);
+    enum
+    {
+        kInvalidIndex = 0xff,
+        kSequenceIndex = kFcfSize,
+    };
+
+    uint16_t GetFrameControlField(void) const;
+    uint8_t FindDstPanIdIndex(void) const;
```

Comment:
Thanks for the explanation. Just curious, is it appropriate to `const_cast` those returned const pointers? The `const_cast` is used inside the class implementation, I think it's safe and efficient. @abtink @jwhui